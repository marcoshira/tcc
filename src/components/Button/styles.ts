import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    color: ${theme.colors.white};
    background: ${theme.colors.primaryColor};
    border-radius: 7px;
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    font-size: ${theme.font.sizes.ltmedium};
    font-weight: 500;
    margin-bottom: 4px;
    transition: ${theme.transitions.faster};

    &:disabled {
      background: ${theme.colors.grey};
      color: ${theme.colors.white};
      border: 2px solid ${theme.colors.grey};
      transition: none;

      &:hover {
        background: ${theme.colors.grey};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.grey};
        cursor: not-allowed;
      }
    }
    &:hover {
      cursor: pointer;
      background: inherit;
      color: ${theme.colors.primaryColor};
    }

    @media ${theme.media.gteMedium} {
      /* width: 33%; */
    }
  `}
`;
