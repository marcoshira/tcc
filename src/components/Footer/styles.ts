import styled, { css } from 'styled-components';

export const Footer = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: white;
    padding: ${theme.spacings.small} 0 ${theme.spacings.medium};

    p {
      margin: 0;
      font-size: 20px;
    }
    a {
      font-size: 18px;
      text-decoration: none;
    }

    @media ${theme.media.gteMedium} {
      padding: ${theme.spacings.small} 8% ${theme.spacings.large};
      flex-direction: row;
      justify-content: space-around;
    }
  `}
`;

export const DevContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin-bottom: 20px;

    @media ${theme.media.gteMedium} {
      margin-bottom: 0px;
      p {
        margin-bottom: 20px;
        font-size: 25px;
      }
      a {
        font-size: 25px;
      }
    }
  `}
`;

export const IconsContainer = styled.div`
  ${({ theme }) => css`
    margin-top: 15px;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    p {
      text-align: center;
    }
    svg {
      color: white;
      height: 40px;
    }
  `}
`;

export const IconsInnerContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    svg {
      color: white;
      height: 40px;
      margin-right: 10px;
    }
    a {
      text-decoration: none;
    }
  `}
`;
