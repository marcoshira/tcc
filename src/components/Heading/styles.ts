import styled, { css } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size ${theme.font.sizes.xlarge}
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, size, uppercase }) => css`
    color: white;
    margin: 0;
    margin-bottom: ${theme.spacings.small};
    padding: ${theme.spacings.large};
    background: ${theme.colors.primaryColor};
    text-align: center;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
