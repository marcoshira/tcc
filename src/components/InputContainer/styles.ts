import styled, { css } from 'styled-components';

type BorderRed = {
  red?: boolean;
  yellow?: boolean;
};

const redBorder = () => css`
  box-shadow: 0px 0px 5px red;
`;

const yellowBorder = () => css`
  box-shadow: 0px 0px 5px yellow;
`;

export const Wrapper = styled.div<BorderRed>`
  ${({ theme, red, yellow }) => css`
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    border-radius: 7px;
    margin-bottom: ${theme.spacings.small};
    width: 100%;
    transition: ${theme.transitions.slowest};

    ${red && redBorder()}

    @media ${theme.media.gteMedium} {
      margin-bottom: ${theme.spacings.large};
    }

    > div {
      > svg {
        position: absolute;
        top: 12px;
        right: 12px;
        width: 32px;
        color: ${theme.colors.primaryColor};

        &:hover {
          color: ${theme.colors.orange};
        }
      }
      &:hover {
        & .tooltiptext2 {
          visibility: visible;
          opacity: 1;
        }
      }
    }

    & .tooltiptext2 {
      visibility: hidden;
      width: 120px;
      background-color: transparent;
      color: gray;
      text-align: center;
      padding: 15px;
      border-radius: 6px;
      border: 1px solid gray;
      /* height: 100%; */
      width: ${theme.frameSizes.small};
      top: 0px;
      left: 101.5%;

      position: absolute;
      z-index: 1;

      opacity: 0;
      transition: opacity 0.3s;

      text-align: center;
    }
    & .tooltiptext2::before {
      content: ' ';
      position: absolute;
      top: 26px;
      right: 100%;
      margin-top: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: transparent gray transparent transparent;
    }
    & .tooltiptext2::after {
      content: ' ';
      position: absolute;
      top: 26px;
      right: 99.6%;
      margin-top: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: transparent white transparent transparent;
    }

    /* Tooltip container */
    &.tooltip {
      position: relative;
      display: inline-block;
      border-bottom: 1px dotted black; /* If you want dots under the hoverable text */
    }

    /* Tooltip text */
    &.tooltip .tooltiptext {
      visibility: hidden;
      width: 120px;
      background-color: red;
      color: white;
      text-align: center;
      padding: 5px 0;
      border-radius: 6px;
      border: 2px solid red;
      height: ${theme.spacings.large};
      width: 100%;
      bottom: 106%;
      left: 50%;
      margin-left: -50%; /* Use half of the width (120/2 = 60), to center the tooltip */

      /* Position the tooltip text - see examples below! */
      position: absolute;
      z-index: 2;

      opacity: 0;
      transition: opacity 0.3s;
    }

    &.tooltip .tooltiptext4 {
      visibility: hidden;
      width: 120px;
      background-color: #fff3ac;
      color: #856404;
      text-align: center;
      padding: 5px 0;
      border-radius: 6px;
      border: 2px solid #fff3ac;
      height: ${theme.spacings.large};
      width: 100%;
      bottom: 106%;
      left: 50%;
      margin-left: -50%; /* Use half of the width (120/2 = 60), to center the tooltip */

      /* Position the tooltip text - see examples below! */
      position: absolute;
      z-index: 1;

      opacity: 0;
      transition: opacity 0.3s;
    }

    /* Show the tooltip text when you mouse over the tooltip container */
    &.tooltip:hover .tooltiptext,
    &.tooltip:hover .tooltiptext4 {
      visibility: visible;
      opacity: 1;
    }

    &.tooltip .tooltiptext::after {
      content: ' ';
      position: absolute;
      top: 100%; /* At the bottom of the tooltip */
      left: 50%;
      margin-left: -9px;
      border-width: 9px;
      border-style: solid;
      border-color: red transparent transparent transparent;
    }

    &.tooltip .tooltiptext4::after {
      content: ' ';
      position: absolute;
      top: 100%; /* At the bottom of the tooltip */
      left: 50%;
      margin-left: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: #fff3cd transparent transparent transparent;
    }
  `}
`;
