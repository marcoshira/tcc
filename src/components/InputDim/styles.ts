import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large} 0 ${theme.spacings.medium};
    margin: ${theme.spacings.medium} 0;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;

    @media ${theme.media.gteMedium} {
      padding: 0 ${theme.spacings.medium} 0 ${theme.spacings.large};
      flex-direction: row;
      justify-content: space-around;
      /* width: 90%; */
      label:first-of-type {
        width: 28%;

        .unit {
          @media ${theme.media.gteMedium} {
            left: 89%;
            /* right: 8px; */
          }
        }
      }

      label:nth-of-type(2) {
        width: 33%;
      }

      label:nth-of-type(3) {
        width: 37%;
      }
    }
  `}
`;
