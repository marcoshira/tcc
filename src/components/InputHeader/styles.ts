import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0px 0px;
    display: flex;
    justify-content: center;

    h2 {
      font-size: ${theme.font.sizes.medium};
      font-weight: 300;
    }
  `}
`;
