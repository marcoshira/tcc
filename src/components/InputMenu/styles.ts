import styled, { css } from 'styled-components';
import { InputMenuProps } from '.';

const frameSize = {
  small: (theme) => css`
    height: ${theme.frameSizes.ltxsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  mediumSmall: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  medium: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.ltxsmall};
    }
  `,
  big: (theme) => css`
    height: ${theme.frameSizes.gtsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.small};
    }
  `,
  huge: (theme) => css`
    height: ${theme.frameSizes.smallMedium};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.gtsmall};
    }
  `,

  spt: (theme, depth) => css`
    height: ${depth < 6
      ? theme.frameSizes.xsmall
      : depth < 8
      ? theme.frameSizes.ltsmall
      : depth < 10
      ? theme.frameSizes.small
      : `${(depth - 10) * 2.5 + 26}rem`};
    @media ${theme.media.gteMedium} {
      height: ${`${(Math.floor(depth / 4) + 1) * 6}rem`};
    }
  `,
};

export const Wrapper = styled.div<InputMenuProps>`
  ${({ theme, size, depth }) => css`
    display: flex;
    justify-content: center;
    height: 0px;
    opacity: 0;

    transition: ${theme.transitions.faster};

    > div {
      display: none;
    }

    &.open {
      ${frameSize[size](theme, depth)};
      opacity: 1;

      > div {
        display: flex;
      }
    }

    &.zero {
      height: 0px;
      opacity: 0;
    }

    @media ${theme.media.gteMedium} {
      ${frameSize[size](theme, depth)};
      opacity: 1;

      > div {
        display: flex;
      }
    }
  `}
`;
