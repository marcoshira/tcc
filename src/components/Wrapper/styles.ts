import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 92%;
    display: flex;
    justify-content: center;
    flex-direction: column;
    position: relative;
    left: 50%;
    transform: translate(-50%, 0);

    @media ${theme.media.gteMedium} {
      width: 85%;
      margin-top: ${theme.spacings.huge};
    }
    @media ${theme.media.gteBig} {
      width: 60%;
      margin-top: ${theme.spacings.huge};
    }
  `}
`;
