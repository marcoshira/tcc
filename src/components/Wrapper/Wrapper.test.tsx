import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Wrapper } from '.';

describe('<Wrapper />', () => {
  it('should render', () => {
    const { container } = renderTheme(
      <Wrapper>
        <p>oiu</p>
      </Wrapper>,
    );

    expect(container).toMatchSnapshot();
  });
});
