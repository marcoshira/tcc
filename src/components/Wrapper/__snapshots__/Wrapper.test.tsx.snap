// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should render 1`] = `
.c0 {
  width: 92%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
}

@media (min-width:768px) {
  .c0 {
    width: 85%;
    margin-top: 5.6rem;
  }
}

@media (min-width:1336px) {
  .c0 {
    width: 60%;
    margin-top: 5.6rem;
  }
}

<div>
  <div
    class="c0"
  >
    <p>
      oiu
    </p>
  </div>
</div>
`;
